File: examples/python/04_error_handling.py
"""Handle daemon API errors without printing secrets."""

from __future__ import annotations

import os

from doge_sdk import DogeApiError, DogeClient


def main() -> None:
    client = DogeClient(
        base_url=os.getenv("DOGE_DAEMON_URL", "http://127.0.0.1:8901"),
        api_token=os.getenv("DOGE_API_TOKEN"),
    )
    try:
        try:
            client.runs.get("run-does-not-exist")
        except DogeApiError as exc:
            print(f"status={exc.status_code}")
            print(f"message={exc}")
    finally:
        client.close()


if __name__ == "__main__":
    main()

File: examples/typescript/04_error_handling.ts
import { DogeApiError, DogeClient } from 'doge-sdk'

const client = new DogeClient({
  baseUrl: process.env.DOGE_DAEMON_URL ?? 'http://127.0.0.1:8901',
  apiToken: process.env.DOGE_API_TOKEN,
})

try {
  await client.runs.get('run-does-not-exist')
} catch (error) {
  if (error instanceof DogeApiError) {
    console.log(`status=${error.statusCode}`)
    console.log(`message=${error.message}`)
  } else {
    throw error
  }
}
